src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCasts.java

```diff
@@ -0,0 +1,109 @@
+/*
+UnnecessaryParentheses
+tokens = (default)EXPR, IDENT, NUM_DOUBLE, NUM_FLOAT, NUM_INT, NUM_LONG, \
+         STRING_LITERAL, LITERAL_NULL, LITERAL_FALSE, LITERAL_TRUE, ASSIGN, \
+         BAND_ASSIGN, BOR_ASSIGN, BSR_ASSIGN, BXOR_ASSIGN, DIV_ASSIGN, \
+         MINUS_ASSIGN, MOD_ASSIGN, PLUS_ASSIGN, SL_ASSIGN, SR_ASSIGN, STAR_ASSIGN, \
+         LAMBDA, TEXT_BLOCK_LITERAL_BEGIN, LAND, LITERAL_INSTANCEOF, GT, LT, GE, \
+         LE, EQUAL, NOT_EQUAL, UNARY_MINUS, UNARY_PLUS, INC, DEC, LNOT, BNOT, \
+         POST_INC, POST_DEC, TYPECAST
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.coding.unnecessaryparentheses;
+import java.util.HashSet;
+import java.util.Arrays;
+public class InputUnnecessaryParenthesesCasts {
+    public void valid1() {
+        int x = 23;
+        int y = 44;
+        float k = 12f;
+
+        int d = ((int) 100f) + 100 * 2 / ((int) 12.5f) + (int) 90f; // 2 violations
+
+        y = (int) (22.2 * 2) / ((int) 8f + 5);
+
+        double arg2 = 23.2;
+        int i = (int) arg2;
+
+        i = ((int) arg2); // violation 'Unnecessary parentheses around assignment right-hand side'
+
+        x = (2 * 2 /((int) k)); // 2 violations
```

Comment:
Please add violation message for both violations.
`// 2 violations below:`

Same for other cases like this

Reply:
done